Composer/packages/adaptive-form/src/components/fields/IntellisenseFields.tsx

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import { FieldProps } from '@bfc/extension-client';
+import { Intellisense } from '@bfc/intellisense';
+import React from 'react';
+
+import { getIntellisenseUrl } from '../../utils/getIntellisenseUrl';
+import { ExpressionSwitchWindow } from '../ExpressionSwitchWindow';
+
+import { JsonField } from './JsonField';
+import { NumberField } from './NumberField';
+import { StringField } from './StringField';
+
+export const IntellisenseTextField: React.FC<FieldProps<string>> = function IntellisenseTextField(props) {
+  const { id, value = '', onChange, uiOptions, focused: defaultFocused } = props;
+
+  const completionListOverrideResolver = (value: string) => {
+    if (value === '') {
+      return <ExpressionSwitchWindow type={'String'} onSwitchToExpression={() => onChange('=')} />;
+    } else {
+      return null;
+    }
+  };
+
+  const scopes = uiOptions.intellisenseScopes || [];
+  const url = getIntellisenseUrl();
+
+  return (
+    <Intellisense
+      completionListOverrideResolver={completionListOverrideResolver}
+      focused={defaultFocused}
+      id={`intellisense-${id}`}
+      scopes={scopes}
+      url={url}
+      value={value}
+      onBlur={props.onBlur}
+      onChange={onChange}
+    >
+      {({ textFieldValue, focused, onValueChanged, onKeyDownTextField, onKeyUpTextField, onClickTextField }) => (
+        <StringField
+          {...props}
+          focused={focused}
+          id={id}
+          value={textFieldValue}
+          onBlur={undefined} // onBlur managed by Intellisense
```

Comment:
make it optional?

Reply:
This is used to override what's decomposed using {...props}.